src/modules/flow/http-client/http-client.c

```diff
@@ -522,7 +525,11 @@ static int
 string_process_data(struct sol_flow_node *node,
     struct sol_http_response *response)
 {
-    char *result = strndup(response->content.data, response->content.used);
+    char *result;
+
+    SOL_HTTP_RESPONSE_CHECK_API(response, -EINVAL);
```

Comment:
No need to check the version here, it's known to be good since that is checked before the packet is created.